packages/cli/src/lib/config.ts

```diff
@@ -0,0 +1,176 @@
+import type {
+  BuildConfig,
+  GenerateOptions,
+  PrebuildOptions,
+  CodeConfig,
+} from "@brandingbrand/code-cli-kit";
+import { program } from "commander";
+import type { PackageJson } from "type-fest";
+
+/**
+ * Configuration object containing various settings.
+ */
+export const config = {
+  /**
+   * @property {Object} __dangerously_access_code - Configuration object for fscode.
+   */
+  __dangerously_access_code: {} as CodeConfig,
+  /**
+   * @property {function} code - Getter function for accessing fscode configuration.
+   */
+  get code() {
+    return this.__dangerously_access_code;
+  },
+  /**
+   * @property {function} code - Setter function for updating fscode configuration.
+   */
+  set code(data: CodeConfig) {
+    this.__dangerously_access_code = data;
+  },
+  /**
+   * @property {Object} __dangerously_access_options - Configuration object for options.
+   */
+  __dangerously_access_options: {} as PrebuildOptions,
+  /**
+   * @property {function} options - Getter function for accessing options configuration.
+   */
+  get options() {
+    return this.__dangerously_access_options;
+  },
+  /**
+   * @property {function} options - Setter function for updating options configuration.
+   */
+  set options(data: PrebuildOptions) {
+    this.__dangerously_access_options = data;
+  },
+  /**
+   * @property {Object} __dangerously_access_generate_options - Configuration object for generate options.
+   */
+  __dangerously_access_generate_options: {} as GenerateOptions,
+  /**
+   * @property {function} generateOptions - Getter function for accessing options configuration.
+   */
+  get generateOptions() {
+    return this.__dangerously_access_generate_options;
+  },
+  /**
+   * @property {function} generateOptions - Setter function for updating options configuration.
+   */
+  set generateOptions(data: GenerateOptions) {
+    // Check if the 'name' property of the data object follows the non-scoped package names format
+    if (!/^[a-z0-9-~][a-z0-9-._~]*$/gm.test(data.name)) {
+      // Throw an error if the name does not match the specified format
+      throw Error(
+        "[GenerateCliCommandError]: name must follow non-scoped package names format"
```

Comment:
why can't we use scoped package name? I can modify the package after generation to scope and it seems to work